Markdown.Sanitizer.js

```diff
@@ -33,8 +33,10 @@
         if (tag.match(basic_tag_whitelist) || tag.match(a_white) || tag.match(img_white)) {
             return tag;
         } else if (tag.match(table_white)) {
-            if (tag.match(/^<table[^>]*>/gi)) {
-                return "<table class=\"challenge-body-table\">";
+            if (tag.match(/^<table(\s+class="[\w-]+")\s*>/gi)) {
```

Comment:
why do you want to return the entire tag if it has class? shouldn't you be returning only the element with class added and removing anything else?